pack/nodejs/provider.ts

```diff
@@ -1376,6 +1376,7 @@ export namespace apiextensions {
           apiVersion: pulumi.Input<string>,
           kind: pulumi.Input<string>
           metadata?: pulumi.Input<outputApi.meta.v1.ListMeta>;
+          [others: string]: any;
```

Comment:
I'm not entirely clear on what this is used for.

Reply:
This is only used one place, right? It doesn't seem worth naming it.

We're using this here because the contract we want to enforce here is: the object must have `apiVersion` and `kind`, but may also contain an arbitrary number of other things.

Reply:
Comments (documentation) are one reason.

Also - should this be `[others: string]: pulumi.Input<any>`.  In practice it doesn't matter (both accept anything), but it is slightly more suggestive of what will happen here (Inputs will be unrolled).

Reply:
@lukehoban ok we can do both.